File: inferno3_database/item.py
from peewee import Model, CharField


class BaseModel(Model):
    class Meta:
        database = database


# the user model specifies its fields (or columns) declaratively, like django
class Item(BaseModel):
    name = CharField(unique=True)
    type = CharField()
    abilities = CharField()
    properties = CharField()
    level = CharField()
    remort = CharField()
    value = CharField()
    weight = CharField()
    armor = CharField()
    effects = CharField()
    prevents = CharField()

File: inferno3-frontend/app/types/item.ts
export interface Item {
  name: string | undefined;
  type: string | undefined;
  level: number | undefined;
  remort: number | undefined;
  value: number | undefined;
  weight: number | undefined;
  armor: number | undefined;
  abilities: string[] | undefined;
  slot: string[] | undefined;
  properties: string[] | undefined;
  effects: object | undefined;
  prevents: string[] | undefined;
  capacity: string | undefined;
  wand: string | undefined;
  mob: string | undefined;
  seller: string | undefined;
  price: string | undefined;
  room: string | undefined;
  area: string | undefined;
}